### PR TITLE
feat: include github actions

### PR DESCRIPTION
**Pull request by kevynfrancobrainstems**

### CODE REVIEW

**Comment by kevynfrancobrainstems** [comment]
@TheSlackOne  In this case, I’m configuring the test run with the following command: pytest --cov=src --cov=tests --cov-report=xml --cov-fail-under=10. I set the coverage threshold at 10% because our current coverage is only 16%. I believe we should create a ticket to develop more tests to increase this coverage.

**Comment by TheSlackOne** [comment]
> @TheSlackOne In this case, I’m configuring the test run with the following command: pytest --cov=src --cov=tests --cov-report=xml --cov-fail-under=10. I set the coverage threshold at 10% because our current coverage is only 16%. I believe we should create a ticket to develop more tests to increase this coverage.

Absolutely. Our first approach was to setup the testing framework with the idea of improving coverage later.

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,5 @@
 websockets
 torch
+pytest
+pytest-cov
+llama-cpp-python
```
Comment (TheSlackOne) [inline requirements.txt]: Watch out with this one. It should compile with the flags described in the README.md. If not, you won't get CUDA acceleration. That why it was installed in isolation with its own flags.

**Review by TheSlackOne — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 1 above)
Comment (kevynfrancobrainstems) [reply requirements.txt]: ok I include a new requirements-text file to support separated cases.

**Review by TheSlackOne — APPROVED** [review]

**Pull request merged**